### PR TITLE
cloud-availability-updater: implement git interactions

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Starts the work on https://github.com/airbytehq/airbyte-cloud/issues/4213
We want to automatically open PRs when a connectors becomes eligible to cloud.
This PR focuses on git interaction logics. PR creation and actual call to the new `deploy_new_connector_to_cloud_repo` will be done in future PRs.

Prior work:
- https://github.com/airbytehq/airbyte/pull/21961 allows to fech a list of connectors eligible to Cloud

Future works:
- Finish https://github.com/airbytehq/airbyte-cloud/issues/4213 by automatically creating PRs
- Define new GitHub Action workflow on `airbyte-cloud` to run the Cloud Availability updater daily. (cf #21918)

## How
Implement the logic to:
- add a connector to the Airbyte-cloud repo definitions mask
- locally regenerate the cloud catalog
- commit / push in a new branch on `airbyte-cloud` repo

## Recommended reading order
https://github.com/airbytehq/airbyte/blob/18abf99a2e93bff89acfa99f0e35f2dc479dbc18/tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py#L24

https://github.com/airbytehq/airbyte/blob/18abf99a2e93bff89acfa99f0e35f2dc479dbc18/tools/ci_connector_ops/tests/test_qa_engine/test_cloud_availability_updater.py#L43

## 🚨 User Impact 🚨
None, the code implemented is not actually called at the moment.

### CODE REVIEW

File: tools/ci_connector_ops/tests/test_qa_engine/test_cloud_availability_updater.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        connector_definition_id="abcdefg",
+        connector_version="0.0.0",
+        release_stage="alpha",
+        is_on_cloud=False,
+        is_appropriate_for_cloud_use=True,
+        latest_build_is_successful=True,
+        documentation_is_available=True
+    )
+    cloud_availability_updater.deploy_new_connector_to_cloud_repo(repo_path, airbyte_cloud_repo, connector)
+    new_branch_name = f"cloud-availability-updater/deploy-{connector.connector_technical_name}"
```
Comment (evantahler) [inline tools/ci_connector_ops/tests/test_qa_engine/test_cloud_availability_updater.py:138]: 👍 connector names shouldn't have any spaces or special chars

**Review by evantahler — APPROVED** [review]
I love code that writes code!
Are there any special permissions you will need to run this on GH actions?

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    return new_branch
+
+def update_definitions_mask(connector: ConnectorQAReport, definitions_mask_path: Path) -> Optional[Path]:
+    with open(definitions_mask_path, "r") as definition_mask:
+        connector_already_in_mask = connector.connector_definition_id in definition_mask.read()
+    if connector_already_in_mask:
+        logging.warning(f"{connector.connector_name}'s definition id is already in {definitions_mask_path}.")
+        return None
+
+    to_append = f"""# {connector.connector_name} (from cloud availability updater)
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:47]: We only need the connector definition id to add it to the mask.
But we usually comment the name of the connector . [e.g ](https://github.com/airbytehq/airbyte-cloud/blob/main/cloud-config/cloud-config-seed/src/main/resources/seed/source_definitions_mask.yaml#L337)

[the remark this answers is not in the document]
Comment (evantahler) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py:47]: Got it!  We should add a note in the PR that this creates then about "...and if you you want to modify the connector (e.g. resourceRequirements) do so before merging this PR"

**Comment by alafanechere** [comment]
> Are there any special permissions you will need to run this on GH actions?
I expect the Octavia PAT to have these super permissions. It's used in [this workflow](https://github.com/airbytehq/airbyte-cloud/blob/main/.github/workflows/generate-cloud-catalog.yml#L1) that commits to master...
If not I'll ask @supertopher 😄

**Comment by supertopher** [comment]
@alafanechere as long as you can commit code you can access secrets which is the only real barrier to GH actions.

they keys mostly have the same permissions, but there are some edge cases 

I think most keys are basically users.  there are a couple that own the org.  If you see a permissions error I think the error message is pretty clear

**Pull request merged**